client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -210,10 +211,18 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
       appUniqueId,
       conf,
       masterClient,
-      () => commitManager.commitMetrics() -> shuffleFallbackCount.sumThenReset(),
+      () => {
+        commitManager.commitMetrics() ->
+          (shuffleCount.sumThenReset(), sumThenResetFallbackCounts())
+      },
       workerStatusTracker,
       registeredShuffle,
       reason => cancelAllActiveStages(reason))
+  private def sumThenResetFallbackCounts(): Map[String, java.lang.Long] = {
+    val fallbackCounts = shuffleFallbackCounts.asScala.filter(_._2 > 0L).toMap
+    shuffleFallbackCounts.clear()
```

Comment:
The clear operation may result in the loss of the shuffleFallbackCount, as it will reset the fallback count that was just added.